Configuration/PyReleaseValidation/scripts/runTheMatrix.py

```diff
@@ -305,6 +334,47 @@ def runSelected(opt):
                       default=None,
                       action='store')
 
+    parser.add_option('--gpu',
+                      help='Use GPU workflow setup if available. It will force --requires-gpu required.',
+                      dest='gpuEnable',
+                      default=False,
+                      action='store_true')
+
+    parser.add_option('--requires-gpu',
+                      help='if GPU is required or not: forbidden (default, CPU-only), optional, required. For relvals, the GPU option will be turned off for optional.',
+                      dest='RequiresGPU',
+                      default='forbidden')
```

Comment:
```suggestion
    parser.add_option('--gpu',
                      help='Enable GPU workflows. Possible options are "forbidden" (default), "required" (implied if no argument is given), or "optional".',
                      dest='gpu',
                      choices=['forbidden', 'optional', 'required'],
                      nargs='?',
                      const='required',
                      default='forbidden',
                      action='store')
```

I would simplify the interface by making `--requires-gpu` redundant:
  - if `--gpu` is not given, `RequiresGPU` defaults to `forbidden` (the current default);
  - if `--gpu` is given without arguments, it implied `required`;
  - `--gpu optional` can be used to specify the third behaviour.

Reply:
Hi @fwyzard 
If we use `choices`, is it possible that `--gpu --wm init` will interpret correctly ( as `--gpu required --wm init`)?

Reply:
Yes, that's how it is supposed to work.
If the next argument starts with a `-` or `--` it is interpreted as a new option, otherwise it is assumed to be an argument o `--gpu`.

Reply:
Wait - I've realised only now that we are still using [optprase](https://docs.python.org/3/library/optparse.html), not [argparse](https://docs.python.org/3/library/argparse.html) - despite it being deprecated since Python 2.7/3.2 ...